src/NHSD.BuyingCatalogue.Ordering.Api/Services/ContactDetailsService.cs

```diff
@@ -1,26 +1,29 @@
-﻿using System;
-using NHSD.BuyingCatalogue.Ordering.Api.Extensions;
+﻿using NHSD.BuyingCatalogue.Ordering.Api.Extensions;
 using NHSD.BuyingCatalogue.Ordering.Api.Models;
 using NHSD.BuyingCatalogue.Ordering.Domain;
-using NHSD.BuyingCatalogue.Ordering.Persistence.Data;
 
 namespace NHSD.BuyingCatalogue.Ordering.Api.Services
 {
     internal sealed class ContactDetailsService : IContactDetailsService
     {
-        private readonly ApplicationDbContext context;
-
-        public ContactDetailsService(ApplicationDbContext context)
-        {
-            this.context = context ?? throw new ArgumentNullException(nameof(context));
-        }
-
         public Address AddOrUpdateAddress(Address existingAddress, AddressModel newOrUpdatedAddress)
         {
             if (existingAddress is null)
                 return newOrUpdatedAddress.ToDomain();
 
-            context.Entry(existingAddress).CurrentValues.SetValues(newOrUpdatedAddress);
+            if (newOrUpdatedAddress is not null)
+            {
+                existingAddress.Line1 = newOrUpdatedAddress.Line1;
+                existingAddress.Line2 = newOrUpdatedAddress.Line2;
+                existingAddress.Line3 = newOrUpdatedAddress.Line3;
+                existingAddress.Line4 = newOrUpdatedAddress.Line4;
+                existingAddress.Line5 = newOrUpdatedAddress.Line5;
+                existingAddress.Town = newOrUpdatedAddress.Town;
+                existingAddress.County = newOrUpdatedAddress.County;
+                existingAddress.Postcode = newOrUpdatedAddress.Postcode;
+                existingAddress.Country = newOrUpdatedAddress.Country;
+            }
```

Comment:
Let's invert the `if` here (to reduce the amount of nested code).

Reply:
Inverted `If` statement to check `Null`